### DISCUSSION TITLE
Alternative touch screen for SSD1963

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [Bodmer/TFT_eSPI#936] [New processor/display support or capability request] [answered]
HI,
I have a 7inch SSD1963 and i've seen here it's not touch screen compatible with esp32,
but i wonder if it's possible to use such remplacement touch screen like that:

![touch screen alt](https://user-images.githubusercontent.com/76820158/104442107-e460c500-5594-11eb-86e1-4abc8a9cb259.png)

I know there is a lot of those touch screens,but why not compatible with drivers that work with espi?

**Comment by Bodmer** [comment, marked as answer]
You will have to find out about the electrical interface and the command/data protocol. It may not have a build in controller. I would not try to use it myself without reading the associated documentation and checking for Arduino support libraries.

The SSD1963 is available with a FT5206 based capacitance based touch screen fitted, these types are much better than the resistance based ones. I expect there is an Arduino library for the FT5206, there is[ example code here](https://www.buydisplay.com/7-tft-screen-touch-lcd-display-module-w-ssd1963-controller-board-mcu) for the display and touch controller.

**Comment by TangerineDreamer** [comment]
Hi ,Bodmer,
Your link doesn't lead to any code,but a selling page,
anyway,i have a 7"inch with a shield that makes it work with arduino mega and arduino due,it's nice and
touch works too,but i prefer to use it with e_spi.I can use it with sep32 but with the touch adress, it can't upload
the sketch,guess you know that.
The touch screen i show you here is one of the many that flows the market recently,i guess nearly all of touch screens
can have it's replacement,that makes me think there must be  somewhere a common driver for them,what do you think?
In the other way,if i can make my 7inch working with a esp32,is it possible to use another mcu (eg:mini d1) to be used
for the touch fonction?

**Reply by Pablo2048** [reply]
Hi @copyCATandpasteDOG 

> Your link doesn't lead to any code,but a selling page,

Scroll down that page and there is section "Tutorial- 8051 Microcontroller" and there you will find the example code...

**Reply by TangerineDreamer** [reply]
Ooops,i didn't scrolled down all the page,thanks to point me there :)

**Comment by Bodmer** [comment]
The touch screen you  have fitted to the SSD1963 display could be used with the ESP32 and TFT_eSPI you just need to find a supporting library. If you examine the working Arduino Mega code you have I think you will be able to convert this to run on the ESP32.
I have the 7" screen in the link I provided, it has the FT5206 fitted. The screen runs OK with TFT_eSPI but I have not used the touch screen, however I expect that there is an FT5206 Arduino library that I could use.
If you wish to use an add on screen then you may have to write your own touch screen driver, you need to examine the screen technical data to work out how to drive it, this is not something I can help with as it is likely to involve many hours of work.
